### DISCUSSION TITLE
Hole punching concept (help needed)

### DISCUSSION THREAD

**Discussion by wpoortman** [magewirephp/magewire#127] [Ideas]
So during the build of Magewire v3, which is far from done, I've been thinking about the whole caching issue developers are walking into when using Magewire on a cached page. Now Magewire could become a core issue solver for this as long as you like / allow XHR requests after page load.

What if Magewire could act as a save-guard for blocks, doesn't even need to be a Magewire component (or maybe it does, I don't know) anyway... what if there is a general implementation loaded on each page checking if there are "hole punched" components on the page. This could be done by some configuration where you can e.g. set a block name which needs to be hole punched, I don't know yet.

Anyway... when it finds a component on the page, it can have a lifetime where if that has exceeded it will only try to reload that component setting a new lifetime and returning fresh html. Would solve a ton of issues where I can't really come up with a solution for the whole FPC problem.

I would like to gather as much feedback as possible to maybe make this possible in the current version of Magewire. All experience and ideas are welcome! Just spit it out 😄

**Comment by Vinai** [comment]
This sounds very similar to the private scope blocks in Magento.
They were deprecated directly after the initial release but still are supported. Not sure if they are still used internally.

I always liked them.
They were deprecated in favor of the section data approach because of the increased number of HTTP requests they caused.

**Reply by wpoortman** [reply]
Is there anything on the internet where I can read more about this? Looked around, but not sure what's legit info.

**Reply by Vinai** [reply]
Probably nothing from Adobe, but there were a number of blogs about it. Oh, there also is https://www.mage2.tv/content/full-page-cache/private-scope-blocks-holepunching/

Let me know if I can set you up with free access if you are interested.

**Comment by JustinElst** [comment]
This might help? https://github.com/elgentos/magento2-clientside-cache

**Comment by Aquive** [comment]
Sounds like a good idea since (full page) caching is very often an issue when developing dynamic things. It would be great to have a nice/elegant and somewhat streamlined solution for this. So very curious to know what would be possible.
